### DISCUSSION TITLE
replicating etc

### DISCUSSION THREAD

**Discussion by pvswie** [psy0rz/zfs_autobackup#187] [Q&A]
Dear All,

I am experimenting on setting up zfs_autobackup using 3 VM test systems :
- controller - this system initiates zfs_autobackup based upon crontab towards 2 external NAS-ses:
- source NAS (XigmaNAS 13)
- destination NAS (XigmaNAS 13)

What I managed so far (reading, trial on error, debugging etc):
- setup a dedicated non-root user on both the source and destination NAS
- assign ZFS rights to each user
- create regular snapshots (04:00, 13:00, 19:00) on the source-NAS
- initial replicated content of source NAS on destination NAS

After replication I ended up with empty folders on the destination NAS while 'zfs list' showed content had arrived. I had to reboot the destination NAS to actually SEE the files. My question is why? What do I need to do such that files show without the reboot?

Peter

**Comment by psy0rz** [comment]
can you show your command and parameters for the replication job?

which version do you use?

**Reply by pvswie** [reply]
Thanks for your interest. 
Please note that this is a test setup, I have not yet specified my intended snapshots etc.

Everything should be latest:

nas-mgr@controller:~$ zfs-autobackup --version
zfs-autobackup v3.1.3 - (c)2021 E.H.Eefting (edwin@datux.nl)

XigmaNAS(2x identical version):
[nas-mgr@xigmanas ~]$ zfs version
zfs-2.1.4-FreeBSD_g52bad4f23
zfs-kmod-2.1.4-FreeBSD_g52bad4f23

The following command is used to perform the replication:
/home/nas-mgr/.local/bin/zfs-autobackup --ssh-source nas-mgr@172.16.1.222 --ssh-target nas-mgr@172.16.1.211 --no-holds --allow-empty --snapshot-format {}-%y%m%d_%H:%M offsite1 pool-0/172.16.1.222

**Comment by psy0rz** [comment]
this is related to #112

try typing zfs mount -a to mount all the zfs-datasets that should be mounted.

**Reply by pvswie** [reply]
zfs mount -a has no effect. Read some more details below on the steps I have taken.
I do have a question, please read at the end.

I tried 'zfs mount -a' after deleting all data on the target NAS:
- I manually created dataset 172.16.1.222 on the target NAS through the XigmaNAS web interface 172.16.1.222. I am unsure I need this at all. See below.
- After the replication command: I do see one additional level, but not the content of it:
/mnt/pool-0/172.16.1.222/pool-0

zfs list
NAME USED  AVAIL     REFER  MOUNTPOINT
pool-0                                 18.6G   249M      108K  /mnt/pool-0
pool-0/172.16.1.222                    18.6G   249M       96K  /mnt/pool-0/172.16.1.222
pool-0/172.16.1.222/pool-0             18.6G   249M       96K  /mnt/pool-0
pool-0/172.16.1.222/pool-0/dataset-0   18.6G   249M     18.6G  /mnt/pool-0/dataset-0
pool-0/172.16.1.222/pool-0/nas-system   168K   249M      108K  /mnt/pool-0/nas-system
pool-0/nas-system                       176K   249M      176K  /mnt/pool-0/nas-system
The content of the source NAS (dataset-0 & nas-system under 'pool-0/172.16.1.222/pool-0') is visible through zfs list but not on the file system

Question:
- The source NAS has one pool (pool-0) and 2 datasets (nas-system & dataset-0). When replicating I get /mnt/pool-0/172.16.1.222/pool-0. The second 'pool-0' is not really needed, if fact it is even confusing due to the naming. What would be the best option to get rid of this: (1) Use '--strip-path 1' or (2) Not create the 172.16.1.211 dataset, call the pool on the sourceNAS 172.16.1.211 and receive towards pool-0.

**Comment by psy0rz** [comment]
what happens if you explicitly mount the dataset, e.g.:
```
zfs mount pool-0/172.16.1.222/pool-0/dataset-0
```

this should mount it under /mnt/pool-0/nas-system
